engine/src/exec-interface-image.cpp

```diff
@@ -344,14 +352,39 @@ void MCImage::GetText(MCExecContext& ctxt, MCDataRef& r_text)
 		}
 
 		if (MCDataCreateWithBytes((const byte_t *)t_data, t_size, r_text))
-			return;
+        {
+            MCactiveimage = this;
+            return true;
+        }
 	}
 	
-	ctxt . Throw();
+	return false;
+}
+
+void MCImage::GetText(MCExecContext& ctxt, MCDataRef& r_text)
```

Comment:
We don't want to always duplicate the image on GetText... Only if we absolutely have to (i.e. if locked). Also, this isn't how you can copy an image object as it is not a 'POD' data type (i.e. one which is just a sequence of POD values in memory).

You need to use the MCImage copy constructor:
```
MCImage *t_new_image = new MCImage(*this)
```